src/posix/net.cpp

```diff
@@ -1399,6 +1404,8 @@ void CurlHttpIO::send_request(CurlHttpContext* httpctx)
         httpctx->posturl.replace(httpctx->posturl.find(httpctx->hostname), httpctx->hostname.size(), httpctx->hostip);
         httpctx->headers = curl_slist_append(httpctx->headers, httpctx->hostheader.c_str());
     }
+    
+#ifndef TARGET_OS_IPHONE
```

Comment:
I don't understand the reason to skip this code. In case there's no IP nor hostname, do we really want let the code continue?

Reply:
Understood: there should be no proxy (iOS doesn't support it yet) and the IP is not available, since we don't solve hostnames with c-ares in advance for iOS anymore, so there's no chance to replace the hostname by the address. However, we want to proceed and send the request directly to the hostname (cUrl will do the lookup).